pilot/pkg/controller/workloadentry/workloadentry_controller.go

```diff
@@ -342,56 +469,34 @@ func (c *Controller) QueueUnregisterWorkload(proxy *model.Proxy, origConnect tim
 	c.mutex.Unlock()
 
 	disconTime := time.Now()
-	if err := c.unregisterWorkload(entryName, proxy, disconTime, origConnect); err != nil {
+	if err := c.unregisterWorkload(entryName, proxy, disconTime, origConnect, autoCreated); err != nil {
 		log.Errorf(err)
 		c.queue.AddRateLimited(&workItem{
 			entryName:   entryName,
 			proxy:       proxy,
 			disConTime:  disconTime,
 			origConTime: origConnect,
+			autoCreated: autoCreated,
 		})
 	}
 }
 
-func (c *Controller) unregisterWorkload(entryName string, proxy *model.Proxy, disconTime, origConnTime time.Time) error {
-	// unset controller, set disconnect time
-	cfg := c.store.Get(gvk.WorkloadEntry, entryName, proxy.Metadata.Namespace)
-	if cfg == nil {
-		// return error and backoff retry to prevent workloadentry leak
-		// TODO(@hzxuzhonghu): update the Get interface, fallback to calling apiserver.
-		return fmt.Errorf("workloadentry %s/%s is not found, maybe deleted or because of propagate latency", proxy.Metadata.Namespace, entryName)
-	}
-
-	// only queue a delete if this disconnect event is associated with the last connect event written to the worload entry
-	if mostRecentConn, err := time.Parse(timeFormat, cfg.Annotations[ConnectedAtAnnotation]); err == nil {
-		if mostRecentConn.After(origConnTime) {
-			// this disconnect event wasn't processed until after we successfully reconnected
-			return nil
+func (c *Controller) unregisterWorkload(entryName string, proxy *model.Proxy, disconTime, origConnTime time.Time, autoCreated bool) error {
+	changed, err := c.changeWorkloadEntryStateToDisconnected(entryName, proxy, disconTime, origConnTime)
+	if err != nil {
+		if autoCreated {
+			autoRegistrationErrors.Increment()
```

Comment:
If it is for this, i feel we can remove it.

Reply:
removed check for `autoCreated` here

but still checking in

```go
	if proxy.WorkloadEntryAutoCreated {
		autoRegistrationUnregistrations.Increment()
	}
```
to avoid confusing metrics